roles/openshift_grafana/tasks/main.yml

```diff
@@ -2,121 +2,142 @@
 - name: Create grafana namespace
   oc_project:
     state: present
-    name: grafana
+    name: "{{ grafana_namespace }}"
 
 - name: Configure Grafana Permissions
   include_tasks: tasks/gf-permissions.yml
-  when: gf_oauth | default(false) | bool == true
-
-# TODO: we should grab this yaml file from openshift/origin
-- name: Templatize grafana yaml
-  template: src=grafana-ocp.yaml dest=/tmp/grafana-ocp.yaml
-  register:
-    cl_file: /tmp/grafana-ocp.yaml
-  when: gf_oauth | default(false) | bool == false
-
-# TODO: we should grab this yaml file from openshift/origin
-- name: Templatize grafana yaml
-  template: src=grafana-ocp-oauth.yaml dest=/tmp/grafana-ocp-oauth.yaml
-  register:
-    cl_file: /tmp/grafana-ocp-oauth.yaml
-  when: gf_oauth | default(false) | bool == true
+
+- name: Pull grafana yaml from openshift/origin
+  get_url:
+    url: "{{ grafana_base_url }}/grafana.yaml"
+    dest: /tmp/grafana.yaml
+
+- name: Create grafana template
+  oc_obj:
+    namespace: "{{ grafana_namespace }}"
+    kind: template
+    name: grafana
+    state: present
+    files:
+    - "/tmp/grafana.yaml"
 
 - name: Process the grafana file
   oc_process:
-    namespace: grafana
-    template_name: "{{ cl_file }}"
+    namespace: "{{ grafana_namespace }}"
+    template_name: grafana
     create: True
-    when: gf_oauth | default(false) | bool == true
 
 - name: Wait to grafana be running
-  command: oc rollout status deployment/grafana-ocp
+  command: oc rollout status deployment/grafana
 
-- name: oc adm policy add-role-to-user view -z grafana-ocp -n {{ gf_prometheus_namespace }}
+- name: Add view role to grafana user
```

Comment:
I would recommend moving this to the top of the task file after the project is created.

Reply:
ok, but just in case the project can't being created, there is no point to add the role
So I've tried create an order for the tasks

Reply:
Ok, thanks.